### PR TITLE
Capital loans page broken on WooCommerce 8.2

### LINKED ISSUE #7531: JavaScript TypeError breaks capital loans page
### Describe the bug

Visiting Payments > Capital shows an error and does not load the expected Capital information.

```
There was an error rendering this view. Please contact support for assistance if the problem persists.
TypeError: Cannot read properties of undefined (reading 'toString')
```
Reported with 6.6.1 with PHP 7.4.33 and 6.6.2 with PHP 8.2.11. No reports of the same error before yesterday. 

From the site logs, the capital information is successfully retrieved from the account but not displayed on the page. 

### To Reproduce

1. Go to Payments > Capital on a site with a capital loan
2. See error displayed

### Additional context

7192208-zen
7195743-zen

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #7531 

#### Changes proposed in this Pull Request

Due to a change in WooCommerce 8.2, `rowsPerPage` is now required on `<TableCard />` components.

This created a broken page for `Capital Loans`. 
image

This fix adds the `rowsPerPage` parameter to allow the component and page to render as expected.

image

Other instances of `<TableCard />` in WooPayments have `rowsPerPage` set and allow dynamic changing using filtering. As the loans page does not have any filtering, I have hardcoded the `rowsPerPage` to the number of loans. 

#### Testing instructions

1. With WooCommerce 8.2 installed and an account with an approved loan
2. Visit the Capital Loans page and confirm it shows as expected.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7569` or branch name `fix/7531` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/7531) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 0d9e714d4fc2e1ab2faaa8124d1ade0d98019dba
- Build time: 2023-10-27 01:49:50 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by Jinksi** [comment]
Would it be worth making `rowsPerPage` a required prop in our self-defined type declaration for `TableCard` as part of this PR @brucealdridge?

https://github.com/Automattic/woocommerce-payments/blob/3b6f89266022ea10bc161025d1293a2baccd51db/client/transactions/declarations.d.ts#L92

**Review by Jinksi — APPROVED** [review]
Looks good @brucealdridge! This PR addresses the immediate issue for merchants, and [correcting our definition of the](https://github.com/woocommerce/woocommerce/blob/57ecec4e99401d92aff58fc5a70a7aee779bb974/packages/js/components/src/table/types.ts#L174) `rowsPerPage` value to a required prop protects us from introducing this error in the future.

I've tested with WC `8.0.3`, `8.1.1` and `8.2.1` and the **Payments → Capital Loans** screen renders without any errors.

As discussed in Slack, this problem may have been avoided if we were using the `TableCard` type definition provided by `@woocommerce/components`. But this would require more substantial effort to upgrade `@woocommerce/components` to a version with the required type definitions (`12.1.0+` - https://github.com/woocommerce/woocommerce/pull/36370).

**Comment by haszari** [comment]
> this problem may have been avoided if we were using the TableCard type definition provided by @woocommerce/components. But this would require more substantial effort to upgrade @woocommerce/components to a version with the required type definitions (12.1.0+)

@anu-rock Is this something Fusion could look at and maybe prioritise? The impact of this specific bug was pretty minor. However it's risky running old versions of dependencies, and more critical flows could potentially break in future.

File: client/capital/index.tsx [hunk 1]
```diff
@@ -218,6 +218,7 @@ const CapitalPage = (): JSX.Element => {
 				totalRows={ loans.length }
 				headers={ columns }
 				rows={ getRowsData( loans ) }
+				rowsPerPage={ loans.length }
```
Comment (Jinksi) [inline client/capital/index.tsx:221]: Note that I also checked that server returns _all_ loans, so pagination is not a concern for this view.

**Pull request merged**